packages/react-reconciler/src/ReactFiberLane.new.js

```diff
@@ -135,8 +136,28 @@ let nextRetryLane: Lane = RetryLane1;
 function getHighestPriorityLanes(lanes: Lanes | Lane): Lanes {
   switch (getHighestPriorityLane(lanes)) {
     case SyncHydrationLane:
+      if (enableUnifiedSyncLane) {
```

Comment:
To simplify this logic a bit, let's just put a single branch at the top of this function:

```js
if (enableUnifiedSyncLane) {
  // This is a constant that includes Default, Sync, etc. But not hydration lanes.
  const pendingSyncLanes = lanes & SyncUpdateLanes;
  if (pendingSyncLanes !== 0) {
    return pendingSyncLanes;
  }
}
```

We can handle the hydration lanes in a similar way in `getBumpedLaneForHydration`:

```js
export function getBumpedLaneForHydration(
  root: FiberRoot,
  renderLanes: Lanes,
): Lane {
  if (enableUnifiedSyncLane) {
    const pendingSyncLanes = lanes & SyncUpdateLanes;
    if (pendingSyncLanes !== 0) {
      return SyncHydrationLane;
    }
  }
}
 
```